src/components/lib/shm_bm/shm_bm.c

```diff
@@ -0,0 +1,193 @@
+#include <shm_bm.h>
+#include <limits.h>
+#include <consts.h>
+#include <string.h>
+
+#define SHM_BM_BITMAP_BLOCK (sizeof (word_t) * CHAR_BIT)
+#define SHM_BM_BITS_TO_WORDS(bits) (round_up_to_pow2(bits, SHM_BM_BITMAP_BLOCK) / SHM_BM_BITMAP_BLOCK)
+#define SHM_BM_ALLOC_ALIGNMENT 1ul << 22 // sm regions are on 4MB boundries
```

Comment:
You can't know that `<<` has a higher associativity than all places this macro is used. Add `()`